app/javascript/gobierto_data/webapp/components/Dataset.vue

```diff
@@ -629,10 +632,12 @@ export default {
       // trigger the modified label:
       // - hides if the new typed query is already stored
       // - shows if the previous query was stored
-      if (this.isQueryStored(sql)) {
-        this.isQueryModified = false;
-      } else if (this.isQueryStored()) {
-        this.isQueryModified = true;
+      if (!this.registrationDisabled) {
+        if (this.isQueryStored(sql)) {
+          this.isQueryModified = false;
+        } else if (this.isQueryStored()) {
+          this.isQueryModified = true;
+        }
```

Comment:
Avoid nested "if" structures as much as you can

```suggestion
  if (!this.registrationDisabled && this.isQueryStored(sql)) {
    this.isQueryModified = false;
  } else if (!this.registrationDisabled && this.isQueryStored()) {
    this.isQueryModified = true;
  }
```